lib/collection/src/shards/local_shard.rs

```diff
@@ -204,7 +205,7 @@ impl LocalShard {
                 thread::Builder::new()
                     .name(format!("shard-load-{collection_id}-{id}"))
                     .spawn(move || {
-                        let mut res = load_segment(&segments_path)?;
+                        let mut res = load_segment(&segments_path, &AtomicBool::new(false))?;
```

Comment:
It seems changes are disabled because here we don't use a stop flag. Why did you decide to add placeholder here instead of actual working?

Reply:
From the issue comments it sounded to me that the focus of this segment loading stopping is the segment optimizer that uses this load_segment function indirectly which can be cancelled, and not LocalShard loading. From where exactly do you expect local shards getting cancelled during loading?

Reply:
@IvanPleshkov https://github.com/qdrant/qdrant/issues/3140#issuecomment-1877541306

Reply:
The idea was to cancel load segments during the optimization, not during the load of shards.